packages/imperative/CHANGELOG.md

```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Imperative package will be documented in this file.
 
+## Recent Changes
+
+- Enhancement: Added support for proxy servers using a proxy http agent. Supports the usage of the environment variables HTTP_PROXY, http_proxy, HTTPS_PROXY, https_proxy. If any of these env variables is set and depending how the Zowe session is configured for http or https it instantiates an appropriate http agent. If the z/OS system uses self-signed certificates then the proxy server must be configured to accept them. If the proxy server itself is configured with self-signed certificates then the user needs to either import these certificates on their workstation, use rejectUnauthorized in their Zowe profile, or use the (not recommended) nodejs variable NODE_TLS_REJECT_UNAUTHORIZED=0. Zowe also looks for the environment variables, NO_PROXY, no_proxy. These work with a simple comma separated list of hostnames that need to match with the hostname of the Zowe profile.
```

Comment:
Thanks for documenting the new behavior thoroughly 👍 

Could we make this long paragraph easier to read by splitting it up into bullet points that explain each environment variable (HTTP_PROXY, HTTPS_PROXY, NO_PROXY)?

Will defer to @anaxceron on this 😋

Reply:
Hi @phaumer -- I did some light editing on the text. Please let me know if we can go ahead w/ the content below for the changelog:

Enhancement: Added support for proxy servers using a proxy http agent. Supports the usage of the environment variables `HTTP_PROXY`, `HTTPS_PROXY` (not case sensitive).
- If any of these environment variables is set and depending how the Zowe session is configured for http or https, the REST client instantiates an appropriate http agent. 
- If the z/OS system uses self-signed certificates then the proxy server must be configured to accept them.
- If the proxy server itself is configured with self-signed certificates then the user needs to either import these certificates on their workstation, use `rejectUnauthorized` in their Zowe profile, or use the (not recommended) nodejs variable `NODE_TLS_REJECT_UNAUTHORIZED=0`.
- Zowe also looks for the environment variable `NO_PROXY`. These work with a simple comma separated list of hostnames that need to match with the hostname of the Zowe profile.

Reply:
I went ahead and updated the changelog witth the suggestion above, thanks @anaxceron !